### PR TITLE
exex: add unit tests for exex manager

### LINKED ISSUE #8692: Exex manager test coverage
### Describe the feature

we have these empty tests

https://github.com/paradigmxyz/reth/blob/6b9623366ed0395def7c1259e3d9f45d23e537c8/crates/exex/exex/src/manager.rs#L484-L497

would be nice to have those tests implemented

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tcoratger**
Should close https://github.com/paradigmxyz/reth/issues/8692

### CODE REVIEW

File: crates/exex/exex/src/manager.rs [hunk 1]
```diff
@@ -481,18 +481,190 @@ impl Clone for ExExManagerHandle {
...
+        assert!(exex_handle.finished_height.is_none());
+
+        // Update the block height via an event
+        event_tx.send(ExExEvent::FinishedHeight(42)).unwrap();
+        let received_event = exex_handle.receiver.recv().await.unwrap();
+        assert_eq!(received_event, ExExEvent::FinishedHeight(42));
+
+        // Check that the block height was updated
+        exex_handle.finished_height = Some(42);
+        assert_eq!(exex_handle.finished_height, Some(42));
```
Comment (shekhirin) [inline crates/exex/exex/src/manager.rs:514]: i don't understand this part, aren't we updating the field by ourself and then it's obviously always equal to the value that we set it to?

File: crates/exex/exex/src/manager.rs [hunk 2]
```diff
@@ -481,18 +481,190 @@ impl Clone for ExExManagerHandle {
...
+            new: Arc::new(Chain::new(vec![block1.clone()], Default::default(), Default::default())),
+        };
+
+        let mut cx = Context::from_waker(futures::task::noop_waker_ref());
+
+        // Send the notification
+        match exex_handle.send(&mut cx, &(22, notification)) {
+            Poll::Ready(Ok(())) => {
+                // The notification should be skipped, so nothing should be sent.
+                // Since nothing is sent, we do not check for received notifications.
```
Comment (shekhirin) [inline crates/exex/exex/src/manager.rs:580]: let's check that the receiver channel is indeed empty?

File: crates/exex/exex/src/manager.rs [hunk 3]
```diff
@@ -481,18 +481,190 @@ impl Clone for ExExManagerHandle {
...
+        }
+
+        // Ensure the notification ID was still incremented
+        assert_eq!(exex_handle.next_notification_id, 23);
+    }
 
     #[tokio::test]
-    async fn capacity() {}
+    async fn test_sends_chain_reorged_notification() {
+        let (mut exex_handle, _, mut notification_rx) = ExExHandle::new("test_exex".to_string());
```
Comment (shekhirin) [inline crates/exex/exex/src/manager.rs:593]: we need to set the finished height for reorg and revert notification tests, so we test that even if the notification block number is lower than the finished height, we still receive the notification

File: crates/exex/exex/src/manager.rs [hunk 4]
```diff
@@ -481,18 +481,190 @@ impl Clone for ExExManagerHandle {
...
+            }
+        }
+
+        // Ensure the notification ID was incremented
+        assert_eq!(exex_handle.next_notification_id, 23);
+    }
 
     #[tokio::test]
-    async fn slow_exex() {}
+    async fn test_slow_exex() {
```
Comment (shekhirin) [inline crates/exex/exex/src/manager.rs:641]: what exactly does this test?

File: crates/exex/exex/src/manager.rs [hunk 5]
```diff
@@ -481,18 +481,190 @@ impl Clone for ExExManagerHandle {
...
+        tokio::time::sleep(delay).await;
+
+        event_tx.send(ExExEvent::FinishedHeight(42)).unwrap();
+        let received_event = exex_handle.receiver.recv().await.unwrap();
+        assert_eq!(received_event, ExExEvent::FinishedHeight(42));
+    }
 
     #[tokio::test]
-    async fn is_ready() {}
+    async fn test_is_ready() {
```
Comment (shekhirin) [inline crates/exex/exex/src/manager.rs:655]: do we need a separate test for that? it seems to be already handled by other tests that send notifications using the handle

**Review by shekhirin — CHANGES_REQUESTED** [review]
Thanks for working on this, we didn't have any tests for the `ExExManager` for a while. Left some comments and suggestions.

File: crates/exex/exex/src/manager.rs (hunk 5 above)
Comment (tcoratger) [reply crates/exex/exex/src/manager.rs]: Agreed, I just followed the test outline that was provided but I actually think that this test is already covered elsewhere and therefore not necessary. I just deleted it.

File: crates/exex/exex/src/manager.rs (hunk 4 above)
Comment (tcoratger) [reply crates/exex/exex/src/manager.rs:641]: I also followed the provided test framework here. I thought the slow test meant to test what happens when the ExEx is slow for some reason. So I added a sleep with a duration of a few milliseconds to simulate this and check that it doesn't break anything and that the event is still received correctly. Tell me what you think, if you want I can also remove this test if you think it is useless.

File: crates/exex/exex/src/manager.rs (hunk 3 above)
Comment (tcoratger) [reply crates/exex/exex/src/manager.rs:593]: Yes better idea, done

File: crates/exex/exex/src/manager.rs (hunk 2 above)
Comment (tcoratger) [reply crates/exex/exex/src/manager.rs]: yes done

File: crates/exex/exex/src/manager.rs (hunk 1 above)
Comment (tcoratger) [reply crates/exex/exex/src/manager.rs]: Yes I forgot to update what I wanted to test: update the finished height via an event. Just fixed.

File: crates/exex/exex/src/manager.rs (hunk 4 above)
Comment (tcoratger) [reply crates/exex/exex/src/manager.rs]: I just deleted it and replaced it with other additional tests to try to have an even more extensive coverage on the functions.

If it suits you, I would say that we can merge first with this level of coverage, even if it means making other follow-up PRs to further increase the coverage little by little, what do you think?

**Review by onbjerg — APPROVED** [review]
lgtm, ty for this

**Review by shekhirin — APPROVED** [review]
LGTM, thank you!

**Pull request merged**